### PR TITLE
Bug 1686683 - Make "YAML Registry Format" reference pages more detailed

### PR DESCRIPTION
**Pull request by brizental**
...And follow new structuring guidelines on it.

This started as just updating a small note on the former "Metrics Parameters" page and developed into a full restructuring and filling of information gaps on the YAML Registry Format documentation. This PR is split into smaller commits to try and help the reviewer :)

### CODE REVIEW

File: docs/user/reference/yaml/metrics.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+{{#include ../../../shared/blockquote-info.html}}
+
+## Note on the naming of these files
+
+> Although we refer to metrics definitions YAML files as `metrics.yaml` throughout Glean documentation
+> this files may be named whatever makes the most sense for each project and may even be broken down
+> into multiple files, if necessary.
+>
+> This does not apply to Firefox Desktop, where all metrics for a component must be declared in a
+> file called _exactly_ `metrics.yaml`.
```
Comment (badboy) [inline docs/user/reference/yaml/metrics.md:23]: Nope, not anymore. They need to be listed [in the index](https://searchfox.org/mozilla-central/source/toolkit/components/glean/metrics_index.py#9-14), so any name should be fine.

File: docs/user/reference/yaml/metrics.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+## File structure
+
+```yaml
+# Schema
+$schema: moz://mozilla.org/schemas/glean/../../reference/metrics/2-0-0
```
Comment (badboy) [inline docs/user/reference/yaml/metrics.md:29]: This should be just `moz://mozilla.org/schemas/glean/metrics/2-0-0` (at least that's what all our metrics.yaml files use).
Looks like it is wrong in `docs/user/user/metrics/adding-new-metrics.md` too.

File: docs/user/reference/yaml/metrics.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+      - http://example.com/path/to/data-review
+    expires: 2019-06-01
+
+  double_click:
+    ...
+```
+
+## Schema
+
+Declaring the schema at the top of a metrics definitions file is advised,
```
Comment (badboy) [inline docs/user/reference/yaml/metrics.md:53]: Afaik it is necessary.

File: docs/user/reference/yaml/metrics.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+
+Categories can contain alphanumeric lower case characters as well as the `.` and `_` characters
+which can be used to provide extra structure, for example `category.subcategory` is a valid category.
+Category lengths may not exceed 40 characters.
+
+Categories may not start with the string `glean`. That prefix is reserved for Glean internal metrics.
+
+Metrics files generated by `glean_parser` will be split by category and each file will be named
+after the category it contains metrics for. See the ["Capitalization"](../../user/metrics/adding-new-metrics.md#capitalization)
+note to understand how the category is formatted before being used as a file name.
```
Comment (badboy) [inline docs/user/reference/yaml/metrics.md:70]: Split into files is only true for Kotlin. For Swift, Desktop, ... we generate a single file

File: docs/user/reference/yaml/metrics.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+A list of bugs (e.g. Bugzilla or GitHub) that are relevant to this metric.
+For example, bugs that track its original implementation or later changes to it.
+
+Each entry should be the full URL to the bug in an issue tracker.
+The use of numbers alone is deprecated and will be an error in the future.
+
+#### `data_reviews`
+
+A list of URIs to any data collection reviews _responses_ relevant to the metric.
```
Comment (badboy) [inline docs/user/reference/yaml/metrics.md:155]:
```suggestion
A list of URIs to any data collection review _responses_ relevant to the metric.
```

File: docs/user/reference/yaml/metrics.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+
+##### Category 4: Highly sensitive data (`highly_sensitive`)
+
+Information that directly identifies a person, or if combined with other data could identify a person.
+Examples include e-mail, usernames, identifiers such as google ad id, apple id, Firefox account,
+city or country (unless small ones are explicitly filtered out), or certain cookies.
+It may be embedded within specific website content, such as memory contents, dumps,
+captures of screen data, or DOM data.
+
+
```
Comment (badboy) [inline docs/user/reference/yaml/metrics.md:248]:
```suggestion
```
Unnecessary newlines

File: docs/user/reference/yaml/pings.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+{{#include ../../../shared/blockquote-info.html}}
+
+## Note on the naming of these files
+
+> Although we refer to pings definitions YAML files as `pings.yaml` throughout Glean documentation
+> this files may be named whatever makes the most sense for each project and may even be broken down
+> into multiple files, if necessary.
+>
+> This does not apply to Firefox Desktop, where all custom pings for a component must be declared
+> in a file called _exactly_ `pings.yaml`.
```
Comment (badboy) [inline docs/user/reference/yaml/pings.md:23]: As above.

File: docs/user/reference/yaml/pings.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+  bugs:
+    - http://bugzilla.mozilla.org/123456789/
+  data_reviews:
+    - http://example.com/path/to/data-review
+```
+
+## Schema
+
+Declaring the schema at the top of a pings definitions file is advised,
+as it is what indicates that the current file is a pings definitions file.
```
Comment (badboy) [inline docs/user/reference/yaml/pings.md:48]: As above

File: docs/user/reference/yaml/pings.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+## Name
+
+Ping names are the top-level keys on pings definitions files. One single definition file
+may contain multiple ping declarations.
```
Comment (badboy) [inline docs/user/reference/yaml/pings.md:53]:
```suggestion
Ping names are the top-level keys on pings definitions files. 
One single definition file may contain multiple ping declarations.
```

https://sembr.org/

File: docs/user/reference/yaml/pings.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+
+["Capitalization"](../../user/metrics/adding-new-metrics.md#capitalization) rules apply to
+ping names on generated code.
+
+{{#include ../../../shared/blockquote-info.html}}
+
+### Reserved ping names
+
+> The names `baseline`, `metrics`, `events`, `deletion-request` and `all-pings` are reserved
+> and may not be used as the name of a custom ping.
```
Comment (badboy) [inline docs/user/reference/yaml/pings.md:69]: Should this mention that the words "custom" and "ping" are also, by default, reject?

File: docs/user/reference/yaml/pings.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+A list of bugs (e.g. Bugzilla or GitHub) that are relevant to this ping.
+For example, bugs that track its original implementation or later changes to it.
+
+Each entry should be the full URL to the bug in an issue tracker.
+The use of numbers alone is deprecated and will be an error in the future.
+
+#### `data_reviews`
+
+A list of URIs to any data collection reviews _responses_ relevant to the metric.
```
Comment (badboy) [inline docs/user/reference/yaml/pings.md:102]:
```suggestion
A list of URIs to any data collection review _responses_ relevant to the metric.
```

**Review by badboy — APPROVED** [review]
r+wc

File: docs/user/reference/yaml/metrics.md (hunk 2 above)
Comment (brizental) [reply docs/user/reference/yaml/metrics.md:29]: Yeah, this whole example is copied from what is already in the book.

File: docs/user/reference/yaml/metrics.md (hunk 3 above)
Comment (brizental) [reply docs/user/reference/yaml/metrics.md:53]: Hm, interesting. It is necessary indeed, also adding a `---` before the schema declaration is also mandatory. I'll change this.

File: docs/user/reference/yaml/metrics.md (hunk 4 above)
Comment (brizental) [reply docs/user/reference/yaml/metrics.md:70]: TIL! It is also true from JavaScript FWIW.

File: docs/user/reference/yaml/pings.md (hunk 10 above)
Comment (brizental) [reply docs/user/reference/yaml/pings.md:69]: I added a note on that in the paragraphs above.

**Pull request merged**